### PR TITLE
CA-148930: mark VDI.parent as deprecated

### PR DESCRIPTION
**Pull request by mseri**
`VDI.parent` field was never set to anything. 
It's not used, SM currently sets `VDI.sm_config["vhd-parent"]` to the parent VDI instead.

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

### CODE REVIEW

File: ocaml/idl/datamodel.ml [hunk 1]
```diff
@@ -6223,7 +6223,7 @@ let vdi =
          field ~in_oss_since:None ~in_product_since:rel_miami ~ty:String ~qualifier:DynamicRO ~default_value:(Some (VString "")) "location" "location information";
          field ~in_oss_since:None ~ty:Bool ~qualifier:DynamicRO "managed" "";
          field ~in_oss_since:None ~ty:Bool ~qualifier:DynamicRO "missing" "true if SR scan operation reported this VDI as not present on disk";
-         field ~in_oss_since:None ~ty:(Ref _vdi) ~qualifier:DynamicRO "parent" "References the parent disk, if this VDI is part of a chain";
+         field ~in_oss_since:None ~ty:(Ref _vdi) ~qualifier:DynamicRO ~lifecycle:[Published, rel_rio, ""; Deprecated, rel_ely, "The field was never used."] "parent" "References the parent disk, if this VDI is part of a chain. This field is deprecated";
```
Comment (thomassa) [inline ocaml/idl/datamodel.ml]: Given that this doesn't hold a reference to the parent, I think the description should not say it does.
Instead we should change the description to something like "Always null. Deprecated."

**Comment by thomassa** [comment]
Looks fine except that the field description still claims that it holds a meaningful reference.

**Comment by thomassa** [comment]
Looks good now. I'm happy for this to be merged (once the last commit has been squashed into the first and the automated tests have passed).

**Comment by mseri** [comment]
Thanks. Done!

**Pull request merged**